test/integration/src/test/java/io/pravega/test/integration/BackUpRecoveryTest.java

```diff
@@ -0,0 +1,1220 @@
+
+        pravegaRunner.shutDownSegmentStoreRunner(); // Shutdown SegmentStore
+        pravegaRunner.shutDownBookKeeperRunner(); // Shutdown BookKeeper & ZooKeeper
+        pravegaRunner.close();
+        log.info("SegmentStore, BookKeeper & ZooKeeper shutdown");
+
+        Thread.sleep(1000);
+        // Start a new segment store and controller
+        PravegaRunner pravegaRunner1 = new PravegaRunner(bookieCount, containerCount);
+        pravegaRunner1.startBookKeeperRunner(instanceId++);
+        log.info("Started bk and zk. ");
+        val bkConfig1 = BookKeeperConfig.builder()
+                .with(BookKeeperConfig.ZK_ADDRESS, "localhost:" + pravegaRunner1.getBookKeeperRunner().getBkPort())
+                .with(BookKeeperConfig.BK_LEDGER_PATH, pravegaRunner1.getBookKeeperRunner().getLedgerPath())
+                .with(BookKeeperConfig.ZK_METADATA_PATH, pravegaRunner1.getBookKeeperRunner().getLogMetaNamespace())
+                .with(BookKeeperConfig.BK_ENSEMBLE_SIZE, 1)
+                .with(BookKeeperConfig.BK_WRITE_QUORUM_SIZE, 1)
+                .with(BookKeeperConfig.BK_ACK_QUORUM_SIZE, 1)
+                .build();
+        this.dataLogFactory = new BookKeeperLogFactory(bkConfig1, pravegaRunner1.getBookKeeperRunner().getZkClient().get(), this.executorService());
+        log.info("Starting segmentstore and controller ");
+        pravegaRunner1.startControllerAndSegmentStore(this.storageFactory, this.dataLogFactory);
+        log.info("Started segment store and controller again.");
+
+        log.info("Starting client to read Events ");
+        // Create the client with new controller.
+        try (val clientRunner = new ClientRunner(pravegaRunner1.getControllerRunner())) {
+            // Try reading all events again to verify that the recovery was successful.
+            readEventsFromStream(clientRunner.clientFactory, clientRunner.readerGroupManager);
+            log.info("Read all events again to verify that segments were recovered.");
+        }
+        FileUtils.deleteDirectory(new File(baseDir.getPath()));
+    }
+
+    /**
+     * Tests the data recovery scenario with transactional writer. Events are written using a transactional writer.
+     *  What test does, step by step:
+     *  1.  Starts Pravega locally with just 4 segment containers.
+     *  2.  Writes {@link #TOTAL_NUM_EVENTS} events in the form of transactions.
+     *  3.  Shuts down controller and segment store
+     *  4.  Restart controller and segment store
+     *  5.  Read transactional events
+     *  6.  Shuts down the controller
+     *  7.  Waits for DurableLog to be entirely flushed to the long term storage.
+     *  8.  Shuts down segment store and bookeeper/zookeeper.
+     *  9.  Starts new instance of bk, zk, store and controller.
+     *  10. Reads all events.
+     * @throws Exception    In case of an exception occurred while execution.
+     */
+    @Test(timeout = 180000)
+    public void testRecoverRestart() throws Exception {
```

Comment:
how is this test different from the one above?

Reply:
removed this test as it is duplicate